### ISSUE TITLE
Anomaly crash throws 500 error code for request at different placesin the code, no error message

### ISSUE THREAD

**Issue by Synida** [Codeception/c3#88]
I spent around a week trying to figure out the issue with this one, so I have to leave some info behind for those that might run into the exact same issue.

So I have around a half thousand functional tests on a project. This project has a module which handles some of the test functions that are using in multiple places, and it sends requests to the API, then checks the results there. I have multiple tests that are using this function, and all of them failed with a 500 error code and no response.
I checked the logs everywhere, but they were all empty, so I started to debug the code, because I thought there might be an issue in it that triggers something related to the coverage generation, which then breaks the code. Interestingly if the tests were executed without the coverage option, then all of them passed without an issue. When I debugged the code I experienced that the code gave up on running in random places. It was most of the time around a specific line, but as I executed the test multiple times sometimes it changed. So I thought there might be a memory issue without error message or maybe a limitation that I reached without an error, but as I increased the memory limit I experienced the same issue and the time limit couldn't be an issue either, because the test failed around 12-16s, which was under the default 30s limit.
Long story short there was a line commented out at the start of the c3.php:
`// $_SERVER['HTTP_X_CODECEPTION_CODECOVERAGE_DEBUG'] = 1;`
where I thought mb I can use this as a debug flag to check out the issue in this part of the code. So after uncommenting the code the test suddenly started passing with the coverage option and rn the tests are running on both locally and on the CI server, so far everything that failed seems to be passing.
